src/main/scala/com/cloudera/sparkts/TimeSeries.scala

```diff
@@ -43,7 +46,8 @@ class TimeSeries(val index: DateTimeIndex, val data: DenseMatrix[Double],
    *   8 pm   5 	4 	      3         10	9 	      8
    *
    */
-  def lags(maxLag: Int, includeOriginals: Boolean): TimeSeries = {
+  def lags(maxLag: Int, includeOriginals: Boolean)(implicit laggedKey: (K, Int) => K)
```

Comment:
One final comment before I merge this: can we actually just have this return a TimeSeries[(K, Int)]?  Apologies for not catching this the first time around.